spring-cloud-starter-stream-processor-httpclient/src/main/java/org/springframework/cloud/stream/app/httpclient/processor/HttpclientProcessorFunctionConfiguration.java

```diff
@@ -99,8 +94,7 @@ else if (properties.getBodyExpression() != null) {
 				return properties.getReplyExpression().getValue(response);
 			}
 			catch (Exception e) {
-				LOG.warn("Error in HTTP request", e);
-				return null;
+				throw new RuntimeException(e.getMessage(), e);
```

Comment:
Why do we need to catch exceptions here at all?
Can't we just rely on the regular execution flow?